tools/missing-test-detector/changed.go

```diff
@@ -66,8 +78,12 @@ func changedFieldsUnder(oldFieldSchema, newFieldSchema *schema.Schema) any {
 			return changedFields(nil, newFieldSchemaElem, true)
 		}
 	}
-	if oldFieldSchema == nil || newFieldSchema.Type != oldFieldSchema.Type || newFieldSchema.Elem != oldFieldSchema.Elem {
```

Comment:
This is where the bug was. The third conditional was comparing pointers where it should be comparing objects.